clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -722,7 +722,8 @@ static void ValidateMultipleRegisterAnnotations(Sema &S, Decl *TheDecl,
 }
 
 static void DiagnoseHLSLRegisterAttribute(Sema &S, SourceLocation &ArgLoc,
-                                          Decl *TheDecl, RegisterType regType) {
+                                          Decl *TheDecl, RegisterType regType,
+                                          const ParsedAttr &AL) {
```

Comment:
I wonder if this might read more cleanly if the last parameter was `bool SpecifiedSpace` or similar.